mbtools/generate_mbz_toc.py

```diff
@@ -48,17 +49,67 @@ def create_toc(mbz_path, md_filepath):
                             current_page = id2page[current_page.next]
 
 
+def create_toc_csv(mbz_path, csv_filepath):
+    moodle_backup = MoodleBackup(mbz_path)
+    activity_list = []
+    for section in moodle_backup.sections():
+        for act in moodle_backup.activities(section.id):
+            if isinstance(act, MoodlePage):
+                for html_elem in act.html_elements():
+                    page_dict = {}
+                    page_dict['section'] = section.title
+                    page_dict['content_id'] = html_elem.get_attribute_values(
+                        "data-content-id")[0]
+                    page_dict['activity_name'] = act.name
+                    page_dict['lesson_page'] = ""
+
+                    activity_list.append(page_dict)
+            elif isinstance(act, MoodleLesson):
+                id2page = {}
+                current_page = None
+                for page in act.lesson_pages():
+                    id2page[page.id] = page
+                    if (page.prev == "0"):
+                        current_page = page
+                while (True):
+                    html_elem = current_page.html_element()
+                    lesson_dict = {}
+                    lesson_dict['section'] = section.title
+                    lesson_dict['content_id'] = html_elem.get_attribute_values(
+                        "data-content-id")[0]
+                    lesson_dict['activity_name'] = act.name
+                    lesson_dict['lesson_page'] = current_page.name
+                    activity_list.append(lesson_dict)
+                    if (current_page.next == "0"):
+                        break
+                    else:
+                        current_page = id2page[current_page.next]
+
+    with open(f"{csv_filepath}/toc.csv", "w") as outfile:
+        content_headers = activity_list[0].keys()
+        result = csv.DictWriter(
+            outfile,
+            fieldnames=content_headers)
+        result.writeheader()
+        result.writerows(activity_list)
+
+
 def main():
     parser = argparse.ArgumentParser(description='')
     parser.add_argument('mbz_path', type=str,
                         help='relative path to the mbz directory')
-    parser.add_argument('md_filepath', type=str,
+    parser.add_argument('output_path', type=str,
                         help='Path/name of the markdown file to be generated')
```

Comment:
Minor: Maybe we can update this to be `Path/name of the output file to be generated` so it encompasses both CSV and Markdown.